Sources/BothamViewController.swift

```diff
@@ -11,31 +11,34 @@ import Foundation
     import UIKit
 
 public class BothamViewController: UIViewController, BothamUI {
-    public var presenter: BothamPresenter! = nil
+    private var _presenter: BothamPresenter? = nil
 
     public override func viewDidLoad() {
         super.viewDidLoad()
-        presenter.viewDidLoad()
+        if self.respondsToSelector(Selector("presenter")) {
+            _presenter = self.performSelector(Selector("presenter")).takeUnretainedValue() as? BothamPresenter
+        }
+        _presenter?.viewDidLoad()
```

Comment:
I'm not sure what you mean.

We could list all properties and test if each of them against `as?
BothamPresenter` and take only the one that fits.

On Mon, Apr 11, 2016 at 5:12 PM, Pedro Vicente Gómez Sánchez <
notifications@github.com> wrote:

> In Sources/BothamViewController.swift
> <https://github.com/Karumi/BothamUI/pull/40#discussion_r59221921>:
>
> >
> >      public override func viewDidLoad() {
> >          super.viewDidLoad()
> > -        presenter.viewDidLoad()
> > +        if self.respondsToSelector(Selector("presenter")) {
> > +            _presenter = self.performSelector(Selector("presenter")).takeUnretainedValue() as? BothamPresenter
> > +        }
> > +        _presenter?.viewDidLoad()
>
> Can't we use as ? BothamPresenter? I wouldn't restrict the name of the
> BothamPresenter used in the view implementation to just presenter.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Karumi/BothamUI/pull/40/files/b469cd2f2e6aeb0c19328dd1a57f079493a9a592#r59221921>
>